axon-server-connector/src/main/java/org/axonframework/axonserver/connector/TopologyChangeListener.java

```diff
@@ -0,0 +1,74 @@
+package org.axonframework.axonserver.connector;
+
+import io.axoniq.axonserver.grpc.control.TopologyChange;
+
+import java.util.function.Consumer;
+
+/**
+ * A functional interface representing a listener that is notified when a {@link TopologyChange} occurs to a specific
+ * {@link AxonServerConfiguration#getContext() context}.
+ * <p>
+ * A {@code TopologyChange} can reflect any of the following {@link TopologyChange#getUpdateType() updates}:
+ * <ol>
+ *     <li>{@code ADD_COMMAND_HANDLER} - A <b>command</b> handler was added by a specific instance to the context the listener is attached to.</li>
+ *     <li>{@code REMOVE_COMMAND_HANDLER} - A <b>command</b> handler was removed from a specific instance for the context the listener is attached to.</li>
+ *     <li>{@code ADD_QUERY_HANDLER} - A <b>query</b> handler was added by a specific instance to the context the listener is attached to.</li>
+ *     <li>{@code REMOVE_QUERY_HANDLER} - A <b>query</b> handler was removed from a specific instance for the context the listener is attached to.</li>
+ *     <li>{@code RESET_ALL} - This signals the connection broke down of a specific instance for the context the listener is attached to.</li>
```

Comment:
Maybe it's a stupid question, but why there is no update about event handlers/processors?

Reply:
Event Handlers nor Event Processors get registered with Axon Server, we only open event streams. And we do that rather frequently when it comes to event sourcing. So, if I had to guess, @MGathier decided against adding such a "persistent stream" of event consumption messages because of that.

Besides that, this feature was introduced as a replacement of the `ConsistentHashChangeListener` that Axon Framework uses for non-Axon Server `CommandBus`.

Then again, having a concrete answer to your question would be good, so I asked @MGathier personally too ;-)
I'll add something to the JavaDoc here to explain.